journal/src/main/java/io/zeebe/journal/file/SegmentedJournal.java

```diff
@@ -120,19 +120,20 @@ public void deleteUntil(final long index) {
         log.debug("{} - Compacting {} segment(s)", name, compactSegments.size());
         for (final JournalSegment segment : compactSegments.values()) {
           log.trace("Deleting segment: {}", segment);
-          segment.compactIndex(index);
           segment.close();
           segment.delete();
           journalMetrics.decSegmentCount();
         }
         compactSegments.clear();
       }
+      journalIndex.deleteUntil(index);
       resetHead(getFirstSegment().index());
     }
   }
 
   @Override
   public void reset(final long nextIndex) {
+    journalIndex.clear();
```

Comment:
This would mean we clear the complete index right? Is this really what we want here? Isn't this call reseting the journal to the given index, like deleteAfter(index-1) and set this index on the writer?

Reply:
reset means deleting all entries in the journal and reset the nextIndex. So, yes. We want to clear the complete index. 

Should we rename the method to make the semantics clear? Any suggestions?

Reply:
yeah it seems I have still issues to understand this method even after discussing several times it with you :laughing: sorry

Reply:
maybe also calling reset on index would help :man_shrugging:

Reply:
I looking at the code and I still don't get it. If we take a look at the writer for example you mentioned that it will reset (clear the journal and set the index to the given one right? Why does it need to read all entries, index them verify the checksum etc. This doesn't make sense to me

Reply:
Journal#reset calls SegmentedJournalWriter#reset which deletes all existing segments and creates a new segment with the given index. 
#reset() in MappedJournalSegmentWriter is different. It is more like re-initializing the writer. Yes, it is confusing. But this was the existing code. I hope that we will fix it as part of rewriting SegmentedJournal implementation.

Reply:
Ok then this will not fix our current bug or? :thinking: The PassiveRole calles directly writer#reset

Reply:
When new journal is integrated to the raft, it can only call journal#reset. So it should not be a problem. 
But when backporting the index should be cleared in the writer.
Btw, only realized now that I fixed it only on the new journal which is not yet used by the raft. Should I also do the fix on the existing one?